### PR TITLE
Feature: Add support for pinning/unpinning tabs

### PR DESCRIPTION
**Pull request by nanashili**
## Summary

This pull request introduces functionality for pinning and unpinning tabs within the TabManager class, ensuring that pinned tabs are moved to the front of the opened tabs list, while unpinned tabs are moved to the back. Additionally, it includes methods to check the pinned status of a tab.

## Changes Made

- Added the pinTab(item:) function to pin a tab and move it to the front of the openedTabs array.
- Added the unpinTab(item:) function to unpin a tab and move it to the back of the openedTabs array.
- Added the isTabPinned(_:) function to check if a tab is pinned.
- Documented the functions using Swift DocC for better understanding and maintainability.

## Motivation

The primary goal of this pull request is to improve the tab management experience by providing clear and consistent behavior for pinning and unpinning tabs. This functionality is necessary to enhance user interaction with the tab bar, ensuring that pinned tabs remain accessible and unpinned tabs do not clutter the view.

## Testing

- Verified that pinned tabs are correctly moved to the front of the openedTabs array.
- Verified that unpinned tabs are correctly moved to the back of the openedTabs array.
- Ensured that the isTabPinned(_:) function accurately reflects the pin status of a tab.

## Screenshots/GIFs

https://cdn.discordapp.com/attachments/1009439782402805800/1245524368788819978/Screen_Recording_2024-05-30_at_01.46.32.MOV?ex=66591076&is=6657bef6&hm=9720637033388222255c39faffc62a4bcb8bf2dceff145bceaab7e78042a0c6a&

## Checklist

Please ensure all of the following are completed before submitting the PR:

- [x] Code has been tested and verified.
- [x] Documentation has been updated to reflect the changes.
- [x] All tests pass successfully.
- [x] Code follows the project's coding guidelines and style.
- [x] The branch is up-to-date with the latest changes from the main branch.
- [ ] Reviewed by at least one other contributor.

## Related Issues

This PR addresses the following issue(s): N/A

## Additional Notes

N/A

### CODE REVIEW

**Comment by aurora-care-bear** [comment]
Thanks for submitting a pull request!
We will review it as soon as possible.

File: AuroraEditor/Base/TabBar/UI/TabBarItem/TabBarItemIcon.swift [hunk 1]
```diff
@@ -30,32 +30,22 @@ extension TabBarItem {
...
-                    // custom buttonStyle. This is an workaround and it works as expected.
                     Button(
                         action: closeAction,
                         label: { EmptyView() }
                     )
                     .frame(width: 0, height: 0)
                     .padding(0)
                     .opacity(0)
-                    .keyboardShortcut("w", modifiers: [.command])
+                    .keyboardShortcut("w", modifiers: [.command, .option])
```
Comment (0xWDG) [inline AuroraEditor/Base/TabBar/UI/TabBarItem/TabBarItemIcon.swift:47]: Why command? in Xcode you close a tab with command + W not command option W

**Review by 0xWDG — CHANGES_REQUESTED** [review]

**Comment by 0xWDG** [comment]
```
2024-06-02T18:07:43.4666330Z <unknown>:0: error: struct declaration cannot close over value 'self' defined in outer scope
2024-06-02T18:07:43.4668210Z /Users/runner/work/AuroraEditor/AuroraEditor/AuroraEditor/Base/Documents/UI/WorkspaceCodeFileView.swift:12:8: note: type declared here
2024-06-02T18:07:43.4669780Z struct WorkspaceCodeFileView: View {
2024-06-02T18:07:43.4670270Z        ^
2024-06-02T18:07:43.4670660Z <unknown>:0: note: 'self' declared here
2024-06-02T18:07:43.4671620Z <unknown>:0: error: struct declaration cannot close over value 'self' defined in outer scope
2024-06-02T18:07:43.4673420Z /Users/runner/work/AuroraEditor/AuroraEditor/AuroraEditor/Base/Documents/UI/WorkspaceCodeFileView.swift:12:8: note: type declared here
2024-06-02T18:07:43.4674870Z struct WorkspaceCodeFileView: View {
2024-06-02T18:07:43.4675340Z        ^
2024-06-02T18:07:43.4675720Z <unknown>:0: note: 'self' declared here
2024-06-02T18:07:43.4676090Z 
```

File: AuroraEditor/Base/TabBar/UI/TabBarItem/TabBarItemIcon.swift (hunk 1 above)
Comment (nanashili) [reply AuroraEditor/Base/TabBar/UI/TabBarItem/TabBarItemIcon.swift:47]: Option click closes all tabs except current one

Comment (0xWDG) [reply AuroraEditor/Base/TabBar/UI/TabBarItem/TabBarItemIcon.swift:47]: Ok, this is breaking the normal Command + W, you should create a separate button for it.

**Pull request closed without merging**